nesta/production/batchables/arxiv/arxiv_elasticsearch/run.py

```diff
@@ -0,0 +1,196 @@
+            if row['citation_count'] is None:
+                row['citation_count'] = 0
+
+            # Extract field of study Level 0 --> Level 1 paths
+            fos = []
+            fos_objs = row.pop('fields_of_study')
+            fos_ids = set(fos['id'] for fos in fos_objs)
+            for f in fos_objs:
+                if f['level'] > 0:
+                    continue
+                fos += [reversed(fos_lookup[(f['id'], cid)])
+                        for cid in split_ids(f['child_ids'])
+                        if cid in fos_ids]
+
+            # Format hierarchical fields as expected by searchkit
+            cats = [(cat['description'], cat['id'].split('.')[0])
+                    for cat in row.pop('categories')]
+            institutes = row.pop('institutes')
+            good_institutes = [i['institute_id'] for i in institutes
+                               if i['matching_score'] > 0.9]
+            countries = set(grid_countries[inst_id]
+                            for inst_id in good_institutes
+                            if inst_id in grid_countries)
+            row['categories'], _, _ = hierarchy_field(cats)
+            row['fos'], novelty0, novelty1 = hierarchy_field(fos)
+            row['countries'], _, _ = hierarchy_field(countries)
+
+            # Pull out international institute info
+            row['has_multinational'] = any(i['is_multinational']
+                                           for i in institutes
+                                           if i['institute_id'] in good_institutes)
+
+            # Generate author & institute properties
+            mag_authors = row.pop('mag_authors')
+            if mag_authors is None:
+                row['authors'] = None
+                row['institutes'] = None
+                row['novelty_of_article'] = 0
+            else:
+                if all('author_order' in a for a in mag_authors):
+                    mag_authors = sorted(mag_authors,
+                                         key=lambda a: a['author_order'])
+
+                row['authors'] = [author['author_name'].title()
+                                  for author in mag_authors]
+                if len(row['authors']) > 10:
+                    row['authors'] = [f"{row['authors'][0]}, et al"]
+
+                gids = [author['affiliation_grid_id']
+                        for author in mag_authors
+                        if 'affiliation_grid_id' in author]
+                row['institutes'] = [grid_institutes[g].title()
+                                     for g in gids
+                                     if g in grid_institutes
+                                     and g in good_institutes]
+                row['novelty_of_article'] = novelty0 + np.log(novelty1+1)
+
+            uid = row.pop('id')
+            #try:
+            _row = es.index(index=es_index, doc_type=es_type,
+                            id=uid, body=row)
+            #except:
```

Comment:
this comment can go now testing is finished?

Reply:
thanks for catching